src/System.IO/src/System/IO/StreamReader.cs

```diff
@@ -155,6 +155,48 @@ public StreamReader(Stream stream, Encoding encoding, bool detectEncodingFromByt
             Init(stream, encoding, detectEncodingFromByteOrderMarks, bufferSize, leaveOpen);
         }
 
+        public StreamReader(string path)
+            : this(path, true)
+        {
+        }
+
+        public StreamReader(string path, bool detectEncodingFromByteOrderMarks)
+            : this(path, Encoding.UTF8, detectEncodingFromByteOrderMarks, DefaultBufferSize)
+        {
+        }
+
+        public StreamReader(string path, Encoding encoding)
+            : this(path, encoding, true, DefaultBufferSize)
+        {
+        }
+
+        public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteOrderMarks)
+            : this(path, encoding, detectEncodingFromByteOrderMarks, DefaultBufferSize)
+        {
+        }
+
+        [System.Security.SecuritySafeCritical]
+        public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteOrderMarks, int bufferSize)
+            : this(path, encoding, detectEncodingFromByteOrderMarks, bufferSize, true)
+        {
+        }
+
+        [System.Security.SecurityCritical]
+        internal StreamReader(string path, Encoding encoding, bool detectEncodingFromByteOrderMarks, int bufferSize, bool checkHost)
```

Comment:
Is `checkHost` even used here? If not, could this constructor be folded into the previous one?